src/snowflake/snowpark/session.py

```diff
@@ -1681,6 +1681,18 @@ def read(self) -> "DataFrameReader":
         supported sources (e.g. a file in a stage) as a DataFrame."""
         return DataFrameReader(self)
 
+    @property
+    def session_id(self) -> int:
+        """Returns an integer that represent the session id of this session."""
+        return self._session_id
+
+    @property
+    def connection(self) -> "ServerConnection":
```

Comment:
We want to expose `self._conn._conn`, which is a `SnowflakeConnection` of Python connector. The ServerConnection is not a public API.

Reply:
Got it, I thought it was self._conn that I should expose